### PR TITLE
Clean up generated code

### PR DESCRIPTION
**Pull request by barronpm**
**Changes**
Updates the generated code to be much cleaner. Split off of #2970 per @mark-monteiro 's request.

### CODE REVIEW

File: Jellyfin.Data/DbContexts/Jellyfin.cs [hunk 1]
```diff
@@ -1,1140 +0,0 @@
...
-         modelBuilder.Entity<global::Jellyfin.Data.Entities.User>()
-                     .HasMany(x => x.Preferences)
-                     .WithOne()
-                     .HasForeignKey("Preference_Preferences_Id")
-                     .IsRequired();
-
-         OnModelCreatedImpl(modelBuilder);
-      }
-   }
-}
```
Comment (mark-monteiro) [inline Jellyfin.Data/DbContexts/Jellyfin.cs:1140]: Why is this deleted?

File: Jellyfin.Data/Entities/Artwork.cs [hunk 2]
```diff
@@ -21,188 +9,194 @@
...
-      /// <summary>
-      /// Required
-      /// </summary>
-      [ConcurrencyCheck]
-      [Required]
-      public byte[] Timestamp { get; set; }
+        public void OnSavingChanges()
+        {
+            RowVersion++;
+        }
```
Comment (mark-monteiro) [inline Jellyfin.Data/Entities/Artwork.cs:194]: Where is the code that calls this?

File: Jellyfin.Data/Entities/Book.cs [hunk 3]
```diff
@@ -21,64 +9,67 @@
...
-      /*************************************************************************
-       * Navigation properties
-       *************************************************************************/
+        /*************************************************************************
+         * Navigation properties
+         *************************************************************************/
 
-      public virtual ICollection<global::Jellyfin.Data.Entities.BookMetadata> BookMetadata { get; protected set; }
+        [ForeignKey("BookMetadata_BookMetadata_Id")]
+        public virtual ICollection<BookMetadata> BookMetadata { get; protected set; }
```
Comment (mark-monteiro) [inline Jellyfin.Data/Entities/Book.cs:68]: Something seems wrong about how the relationships are configured. Normally you don't need to use the `[ForeignKey]` attribute at all if the models are set up correctly.

For example, in this case I would expect the `BookMetadata` class to have `BookId` and `Book` navigation properties, in which case the `[ForeignKey]` attribute is not needed because the relationship is specified by convention. Maybe there were issues getting this to work because the navigation properties on the other side of the relationship are missing?

File: Jellyfin.Data/Entities/Artwork.cs [hunk 4]
```diff
@@ -21,188 +9,194 @@
...
-            int value = _Id;
-            GetId(ref value);
-            return (_Id = value);
-         }
-         protected set
-         {
-            int oldValue = _Id;
-            SetId(oldValue, ref value);
-            if (oldValue != value)
+    [Table("Artwork")]
```
[the remark this answers is not in the document]
Comment (barronpm) [reply Jellyfin.Data/Entities/Artwork.cs:12]: No, you can see how it was done before those annotations in the deleted class `Jellyfin.cs`

File: Jellyfin.Data/DbContexts/Jellyfin.cs (hunk 1 above)
Comment (barronpm) [reply Jellyfin.Data/DbContexts/Jellyfin.cs:1140]: Replaced by `JellyfinDb`

File: Jellyfin.Data/Entities/Artwork.cs (hunk 2 above)
Comment (barronpm) [reply Jellyfin.Data/Entities/Artwork.cs:194]: 🤷 @EraYaN wrote that

File: Jellyfin.Data/Entities/Book.cs (hunk 3 above)
Comment (barronpm) [reply Jellyfin.Data/Entities/Book.cs:68]: cc @EraYaN

File: Jellyfin.Data/DbContexts/Jellyfin.cs (hunk 1 above)
Comment (mark-monteiro) [reply Jellyfin.Data/DbContexts/Jellyfin.cs:1140]: Right, but I think it would be better to move it instead of delete and add, so that we can get a proper diff of the changes made to it. If you do a `git mv` for it in this PR then in #2970 we'll be able to see a diff of the changes made.

File: Jellyfin.Data/Entities/Artwork.cs (hunk 2 above)
Comment (EraYaN) [reply Jellyfin.Data/Entities/Artwork.cs:194]: It's in the DbContext. It calls all these on all objects that are modified.

File: Jellyfin.Data/Entities/Book.cs (hunk 3 above)
Comment (EraYaN) [reply Jellyfin.Data/Entities/Book.cs:68]: This is to specify the name of the foreign key, even if there is no navigation property. But if you want to add all of them, feel free. This is just a direct port of the info the generator put in, and it works.

**Comment by barronpm** [comment]
I see that it's still not showing changes to the `DbContext`, so I'll summarize here:

- Remove unnecessary explicit namespace usage

- Remove unnecessary fluent API calls specifying the db schema

File: Jellyfin.Data/Entities/Book.cs (hunk 3 above)
Comment (mark-monteiro) [reply Jellyfin.Data/Entities/Book.cs:62]: Fair enough, can be fixed later if necessary. I wonder why the generator didn't add the navigation properties to be begin with though :/

File: Jellyfin.Server.Implementations/JellyfinDb.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            OnModelCreatedImpl(modelBuilder);
+        }
+
+        public override int SaveChanges()
+        {
+            foreach (var entity in ChangeTracker.Entries().Where(e => e.State == EntityState.Modified))
+            {
+                var saveEntity = entity.Entity as ISavingChanges;
+                saveEntity.OnSavingChanges();
+            }
```
Comment (mark-monteiro) [inline Jellyfin.Server.Implementations/JellyfinDb.cs:110]: @EraYaN The abstraction `ISavingChanges` seems a bit wrong here. The name implies that the callback would be executed when created or modified but this implementation only executes the callback when modified.

What I would have done is name the interface something like `IHasConcurrencyToken`, include the `RowVersion` property as part of the interface and then call `RowVersion++` here instead of copying that code into every model class. Thoughts on that approach? Could even be used alongside the `ISavingChanges` interface if we want to keep that around for other reasons.

Opening this discussion here since this is where the diff first shows up, but I'm fine to deal with this in a separate PR.

File: Jellyfin.Data/Entities/Book.cs [hunk 6]
```diff
@@ -1,84 +1,69 @@
...
-   {
-      partial void Init();
+    public partial class Book : LibraryItem
+    {
+        partial void Init();
 
-      /// <summary>
-      /// Default constructor. Protected due to required properties, but present because EF needs it.
-      /// </summary>
-      protected Book(): base()
```
[the remark this answers is not in the document]
Comment (barronpm) [reply Jellyfin.Data/Entities/Book.cs:31]: It’s implicit

[the remark this answers is not in the document]
Comment (mark-monteiro) [reply Jellyfin.Data/Entities/Book.cs:31]: huh, yeah makes sense TIL

File: Jellyfin.Server.Implementations/JellyfinDb.cs (hunk 5 above)
Comment (EraYaN) [reply Jellyfin.Server.Implementations/JellyfinDb.cs:110]: Right now these PRs are such a mess with split up changes, that it might be better to just merge and then open a new PR.
